### ISSUE TITLE
FlowLine - How to get the exact location of the current drawing step in `color` function?

### ISSUE THREAD

**Issue by downace** [Viglino/ol-ext#1116]
Looks like the `color` function is called only for visible segments, but the `step` value is always in [0,1] range, so it's not consistent and can't be used to identify the exact location on the map.

Reproduction: https://codesandbox.io/p/sandbox/hnvgwt

<details>
<summary>Reproduction GIF</summary>

![1](https://github.com/user-attachments/assets/da2604e0-5aca-4993-871f-ea755c35ec6c)
</details>

As a workaround I could use `visible: false` (FLowLine) and `updateWhileAnimating: true` (VectorLayer), but

> the setting will have a performance impact for large amounts of vector data.

**Comment by downace** [comment]
Btw, when using `visible: false` without `updateWhileAnimating: true`, it results in a weird behaviour when zooming:

<details>
<summary>GIF</summary>

![2](https://github.com/user-attachments/assets/907a3cc3-ff95-4d42-a535-f4c43dfa6397)
</details>

**Comment by Viglino** [comment]
Yes, it is an optimization to calculate only on visible part. `visible: false` will use the whole geometry but is less efficient...
